### DISCUSSION TITLE
Ideas on how to create a node.js client

### DISCUSSION THREAD

**Discussion by dodiego** [apache/age#35] [Ideas]
Hello everyone,

After reading the pdf docs and thinking about how we should proceed to create a node.js client, I have a few ideas:

- Create a wrapper around [the official node.js client](https://www.npmjs.com/package/pg) with a builtin-option to parse the graph array from the results to an equivalent JSON structure.
- Add [the official node.js client](https://www.npmjs.com/package/pg) as a peer-dependency of this package
- Write using TypeScript so we may have better developer experience out of the box
- Call the npm package [apache-age](https://www.npmjs.com/package/apache-age) :)

If you believe these are good ideas I'll start writing the client right away.

Also, if you think there is room for improvements, let's talk about it!

**Comment by emotionbug** [comment]
I agree that it's a good idea.
but, I'm not sure what's good between wrapper or settable function like my PR #37 .

I prefer the wrapper.
because, I am worried that if I we AGE-related types to pg-types and make connections of multiple servers in one Node.js app,
there will be problems related to the type.

https://github.com/brianc/node-pg-types/blob/c4f8c120d245657f03b0972fdf923b9e296a9eaf/index.js#L9

**Reply by clayrisser** [reply]
Already done. I used the code from Apache Viewer.

https://www.npmjs.com/package/apache-age-client

https://gitlab.com/risserlabs/community/node-apache-age-client

https://github.com/clayrisser/node-apache-age-client

**Comment by mbledkowski** [comment]
Is it possible to use Apache AGE with opencypher driver (like neo4j-driver)?

**Comment by clayrisser** [comment]
@mbledkowski I don't think so at the moment because it has to convert the query into a sql query. Also some features are neo4j are not fully supported yet in apache age.

I suggest using apache-age-client at the moment. Feel free to submit PR's or suggestions on how we can improve the api.

https://www.npmjs.com/package/apache-age-client
